### DISCUSSION TITLE
Label names?

### DISCUSSION THREAD

**Discussion by rsommer** [burningalchemist/sql_exporter#324] [Ideas]
Hi,

just a thought about the current label set when using `jobs` for targeting multiple logical databases in a single server. Currently the target's key is exported as `instance` label. As of the prometheus docs, `instance` reflects `host:port` of the scraped target - which should(?) be the exporter host and it's port and is set by prometheus itself if not present. Wouldn't something like `database` or `dbname` be a more suitable label name in this case? The final data in prometheus depends on whether `honor_labels` is `true` or `false`  otherwise - so with the same exporter config the final data may differ if `instance` is used as label. There was a comment regarding labels and upcoming changes in v0.13 - maybe this will be addressed with the next release?

**Comment by burningalchemist** [comment]
Hey @rsommer, yeah I totally agree. 👍 `instance` was originally set in the project many years ago, and I also think it's not ideal and a bit misleading. My plan is replace it with `target` instead with a prior announcement of course. The thing is that the target is not always a database (although, everything technically can be called a database 😆) - we can query `json`, `csv` or `pgbouncer`, for example. That's why to be more target agnostic I want to rename it to `target`. What do you think?

**Reply by rsommer** [reply]
Hi @burningalchemist , `target` would be fine as well - I'm not sure if this should be configurable after all? I think it depends on how people use sql_exporter. If you deploy an instance along every server instance maybe `database` would be the correct name (because it is highly likely that logical databases are addressed). If someone deploys a single export targeting multiple different hosts with different databases, maybe `target` is more accurate?

**Reply by burningalchemist** [reply]
Yeah, fair enough. I wanted to use `target` everywhere for consistency (between a single target or multiple targets). Let me think about it, we can indeed make it configurable in the end.

**Reply by keithf4** [reply]
I second the `target` label for these as well. Matches what's in the job configuration names and avoids confusion with `instance`.

**Comment by burningalchemist** [comment]
The PR is created. We set target as a default label (instead of instance) which can be adjusted with --config.target-label=<name> parameter.
